packages/blob/README.md

```diff
@@ -285,79 +275,57 @@ export default function UploadForm() {
 ```
 
 ```ts
-// /app/api/generate-blob-client-token/route.ts
+// /app/api/upload/avatars/route.ts
 
-import {
-  generateClientTokenFromReadWriteToken,
-  type GenerateClientTokenOptions,
-} from '@vercel/blob';
+import { handleBlobUpload, type HandleBlobUploadBody } from '@vercel/blob';
 import { NextResponse } from 'next/server';
 
 export async function POST(request: Request): Promise<NextResponse> {
-  // On a real website, this route would be protected by authentication, see: https://nextjs.org/docs/pages/building-your-application/routing/authenticating
-  // Here, we accept the `pathname` from the browser, but in some situations, you may even craft the pathname
-  // based on the authentication result
-  const body = (await request.json()) as { pathname: string };
-
-  const timestamp = new Date();
-  timestamp.setSeconds(timestamp.getSeconds() + 60);
-  return NextResponse.json({
-    clientToken: await generateClientTokenFromReadWriteToken({
-      ...body,
-      onUploadCompleted: {
-        callbackUrl: `https://${
-          process.env.VERCEL_URL ?? ''
-        }/api/file-upload-completed`,
-        metadata: JSON.stringify({ userId: 12345 }),
+  const body = (await request.json()) as HandleBlobUploadBody;
+
+  // The request will run at least 2 times:
+  // - the first time when generating a client token, the body will contain the {type: "blob.generate-client-token"} object.
+  // - the second time once the file upload completed, the body will contain the {type: "blob.upload-completed"} object. This request will retry for 5 times in case the response returns an error status code.
```

Comment:
```suggestion
  // Uploading from browsers is a three-step process:
  // - First, a specific client token is generated by this current route and sent to the browser. The request body will contain a `{ type: "blob.generate-client-token", ... }` object.
  // - Second, the file is uploaded to Vercel Blob directly from the browser, using the client token. The file content doesn't go through this route.
  // - Third, a webhook is called on this route (`onUploadCompleted`). The request body will contain a `{ type: "blob.upload-completed", ... }` object. This webhook will be retried five times in case your route doesn't reply with a 200.
```